### ISSUE TITLE
http_request_factory.js -> _getEncodedEndpoint does not encode querystring array values correctly

### ISSUE THREAD

**Issue by krisarsov** [HandCash/handcash-connect-sdk-js#30]
Hello,

While testing the:
`account.profile.getPublicProfilesByHandle` via:
```
const dotenv = require('dotenv');

// Load environment variables from .env
dotenv.config();

(async function() {
    const {HandCashConnect} = require('@handcash/handcash-connect');
    const handCashConnect = new HandCashConnect({
        appId: process.env.HC_APP_ID,
        appSecret: process.env.HC_APP_SECRET,
    });

    const account = handCashConnect.getAccountFromAuthToken(process.env.HC_AUTH_TOKEN);

    const users = ['antoanetavp', 'yavor', 'karsov']
    const userProfiles = await account.profile.getPublicProfilesByHandle(users);
    console.log(userProfiles);
})();
```

I got the following error:
```
{
  baseURL: 'https://cloud.handcash.io',
  url: '/v1/connect/profile/publicUserProfiles?aliases%5B%5D=antoanetavp%2Cyavor%2Ckarsov',
  method: 'GET',
  headers: {
    'oauth-publickey': '03b50361a16323fc82052d7644ad05f2038d058f350f80f13512a838262a00751b',
    'oauth-signature': '3045022100b8612f90add908233e5b0d82db2cde7b13cfcf76672d50157be1d0879ec0c32402201d5de4db25a07ecfadf8af32516dbf10db86d7a90077faf62bcc8c5a88a7c022',
    'oauth-timestamp': '2022-09-28T11:48:41.710Z',
    'app-secret': 'd63e32ee1e425a143c7a777b90fe1d1d8cc694bbbb8c5982d54691e8dde54a81'
  },
  data: '',
  responseType: 'json'
}
/Users/user/repo/node_modules/@handcash/handcash-connect/src/api/handcash_connect_service.js:103
      return Promise.reject(new HandCashConnectApiError(
                            ^

HandCashConnectApiError: "aliases[0]" with value "antoanetavp,yavor,karsov" fails to match the required pattern: /^[\w\-_.]{3,50}$/
    at handleApiError (/Users/user/repo/node_modules/@handcash/handcash-connect/src/api/handcash_connect_service.js:103:29)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/user/repo/app.js:21:26 {
  httpStatusCode: 400,
  info: {
    path: [ 'aliases', 0 ],
    type: 'string.pattern.base',
    context: {
      regex: {},
      value: 'antoanetavp,yavor,karsov',
      label: 'aliases[0]',
      key: 0
    }
  }
}
```

When passed an array of values for a GET parameter e.g. `'aliases[]': aliases,`
**_getEncodedEndpoint** encodes the URL to: `?aliases[]=antoanetavp,yavor,karsov`
instead it must produce `?aliases[]=antoanetavp&aliases[]=yavor&aliases[]=karsov`

**Comment by rseibane** [comment]
Fixed in #31 by yourself 😄

**Comment by krisarsov** [comment]
> Fixed in #31 by yourself 😄

Yeah 😄
